### ISSUE TITLE
Visualize waypoints and caches belonging together (online/offline)

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#14547]
### Tell us your idea!

Base issue is the same as written in #14502: If we do several multi-caches in the same area all waypoints have the same icon (puzzle piece). This way it is not possible to differentiate the waypoints for the different caches. Example (from #14502):

![waypoints](https://github.com/cgeo/cgeo/assets/85786015/624e4467-977c-46ce-915e-c12159ec998f)

The solution proposed and implemented in #14502 is good but works only for caches formerly stored offline and with custom cache symbols assigned. I want a solution which also works for pure online caches (e.g. on live map) and without having to change the cache symbol.

### What solution would you suggest?

Introduce a new setting in "Map Options" named "Display group marker for caches with waypoints". If this setting is chosen then on the map:
* For each cache with waypoints a hash number between 0-15 is calculated from coordinates and geocode e.g. like this: `(ASCII-Code-of-last-char-of-geocode + latituteAsE6 + longituteAsE6) % 16 `  (or something similar)
* Out of a fixed set of 16 symbols one is chosen for the cache
* This symbol is displayed both on the right side of the cache symbol and the belonging waypoints

That way it would be easy to see which waypoints belong to which cache. This would also work for online caches.

Using 16 symbols is just exemplary, we could agree on another (larger or smaller) set

### Is there an alternative?

Not provide the feature

**Comment by eddiemuc** [comment]
Update: symbol is just transiently chosen for map display. Choice won't be persisted on the cache.

**Comment by eddiemuc** [comment]
Would look something like this:
![image](https://github.com/cgeo/cgeo/assets/6909759/9877f083-8464-4042-9f2f-03474cb32b02)

**Comment by eddiemuc** [comment]
Draft PR: #14553

**Comment by fm-sys** [comment]
Hm, this looks IMHO kinda ugly. Why not changing the white waypoint background into different colors?

**Comment by ztNFny** [comment]
In the can 2 weeks ago we agreed to use a marker in the top left to indicate this, something I'm working on in #14502
I find it disappointing that the shared results of that call are now being ignored - why do i even bother spending lots of time finetuning UI components just to have someone ignore that decision sand implement a solution that clashes with existing functionality and goes against that group-decision?!

**Comment by eddiemuc** [comment]
> In the call 2 weeks ago we agreed to use a marker in the top left to indicate this, something I'm working on in #14502 I find it disappointing that the shared results of that call are now being ignored - why do i even bother spending lots of time finetuning UI components just to have someone ignore that decision sand implement a solution that clashes with existing functionality and goes against that group-decision?!

I agree that the group decision should still stand and the implem,entation is still valid. 
This issue adresses a different topic -> namely displaying grouping for not-offline-stored-caches and w/o having to change cache icon

**Comment by eddiemuc** [comment]
> Hm, this looks IMHO kinda ugly. Why not changing the white waypoint background into different colors?

Also here I agree with @ztNFny : if there was a group decision against other background colors (which I assume was the case in team meeting - I was not there) I won't open again this discussion

**Comment by ztNFny** [comment]
> I agree that the group decision should still stand and the implem,entation is still valid. 
> This issue adresses a different topic -> namely displaying grouping for not-offline-stored-caches and w/o having to change cache icon

Your implementation introduces a GUI which goes against that decision - a second, completely different way to show that WPs belong together, and this way even conflicts with existing GUI.

How about you just wait for #14502 to be implemented and then if improvements are still needed it's done in top of that - instead of this conflicting approach.

**Comment by eddiemuc** [comment]
I fail to see a conflict here. The solution discussed in #14502 fulfills the need of showing waypoints belonging to caches for offline stored caches with custom icon (only). The solution also introduces some nice bonus-features not initially requested but still valid (e.g. numbering of linear waypoints for labs). The solution is valid and should be implemented.

This issue is specifically about grouping waypoints to caches for not offline stored caches and w/o having to set a cache icon. Two different things.

GUI design can't be a problem here too because the usage of icons on the sides of cache icon is already introduced, and the solution follows the group decision to not introduce other GUI elemnts like e.g. different ackground colors.

Thanks to a well-designed `MapMarkerUtils` class (kudos for that), both implementations are super-easy too. Do you want me to implement #14502 in parallel?

Plus the feature in this issue shall be optional, so only people wanting it will see it.

I really fail to see where the problem is here.

**Comment by ztNFny** [comment]
Decision: put it as to left MARKER - implementation: put it as middle right ICON

Middle right position is already taken, your mixing 2 completely unrelated and to a user not differantiatble icons in the same spot. The only way to kbow what the icon is in any given instance is to debug the code

14502 will put it in the top left for caches with a custom icon set, this implemention puts it in the middle right for caches without custom icon set. So 2 completely different representations for exaclty the same thing.

If you fail to see any conflict there - there's 3! - then i can't help you ...

**Comment by eddiemuc** [comment]
I still don't see the conflict here. The markers put onto a cache icon are complex as is, I agree, but I assume there's some manual page explaining this. For example I didn't know that list items are displayed on both left and right side of icon (found this out only by code debugging). Wherever this is documented towards the end user, it can be updated there as well.

A decision not to do any auto-add of markers (ever?) and not add any settings is not known to me. I stand back when this was decided by the team. Was this maybe simply not documented before?.

In any case, this topic seems to be very controversial to you. I will leave it be for now. Maybe something for next team meeting?

**Comment by ztNFny** [comment]
"it's already complex so let's just fuck it up completely and put icons for random stuff in random places" ...

**Comment by ztNFny** [comment]
Before your even started working on this i asked you to hold back as it'd for sure conflict with my ongoing work - which it does - and it conflicts with existing implementations. That you personally didn't know about them, too bad.

If we take a group decision to implement a big feature and then someone comes along and implements a conflicting, worse solution then I'm obviously unable to rely on any decision taken and this unable to work in this environment for anything that takes more that 5 seconds to implement.

**Comment by eddiemuc** [comment]
As said, I will stand back and for sure won't merge anything. I will also respect any team decisions taken, I need to know them of course :-)

As for the PR: don't worry, implementation was a fast job. Something done while waiting for our CI to check on other PRs, so no time wasted. If it is thrown away then it is thrown away.

**Comment by moving-bits** [comment]
Hey everybody, I just entered reading this discussion here and in other issues/PR related to markers' topic this morning. I have to admit that I find the language used currently a bit disturbing/aggressive, and I ask everybody involved here to calm down a bit. We are a team of volunteers, everybody does his/her work in their spare time, and I guess that nobody willingly tries to cancel out or break some other's work. We may have different views on things, and that's ok, and we should respect each other's views and work.

That being said, I remember last team meetings results on this topic to be to start with the things described by @ztNFny in https://github.com/cgeo/cgeo/issues/14502#issuecomment-1676448955 (which is why I did not add this to my little minutes explicitly but referenced it). We had some more ideas, and we already knew at that time, that this would not solve all cases, but decided to start with that feature set, see how it works out in the nightlies, and move on further from that. Thus I propose to refrain from parallel PR for now, but allow discussion of the current proposal as we may have missed relevant use cases in our dev meeting.

Just my 2 cents.

**Comment by moving-bits** [comment]
This issue came up again on support recently, see #17005:

> I remember having that discussion quite some time ago, but could not find the issue any more... (IIRC we had either auto-color coding or connecting waypoints of the same cache by lines, but I'm not sure.)
> 
> User request on support (ticket 323429):
> 
> "Thanks for all the amazing work you do! We were wondering if it would be possible to make waypoints of different caches easier to tell apart by color.
> 
> Sometimes an area has several (multi)geocaches in that same area, sometimes pretty close together. When we are walking them it can be a bit of a hassle to tell which waypoint belongs to which geocache. So perhaps it is possible to implement some kind of color coding them?"

Due to @SammysHP's referral to the current implementation of connecting lines for waypoints, I mentioned this to the user. They tested it, and replied:

"Thank you for the fast reply, we tried this but it isn't very helpful. We can indeed connect waypoints from a cache but if more multi caches are around we still can't see which waypoint belongs to which cache. Also when going back a screen it didn't remember the route so that would be very complicated and repetitive work.
We still hope a color coding might be possible."

For now, the request for color coding is documented here, and additionally I will inform the user about how to create an individual route - maybe this will be sufficient already.

**Comment by eddiemuc** [comment]
I remember this was a long and tiresome discussion back then. I am not ready to start it over another time. But I am supportive for any color coded scenario. I would really like such a feature.
